src/App.js

```diff
@@ -10,40 +15,165 @@ class App extends React.Component {
       // Set default value of card deck to new shuffled deck
       cardDeck: makeShuffledDeck(),
       // currCards holds the cards from the current round
-      currCards: [],
+      round: 0,
+      player1: { winGame: 0, score: 0, result: "", currCard: {} },
+      player2: { winGame: 0, score: 0, result: "", currCard: {} },
+      restart: "",
+      dealButton: <button onClick={this.changeResult}>Deal</button>,
     };
   }
 
   dealCards = () => {
     // this.state.cardDeck.pop() modifies this.state.cardDeck array
-    const newCurrCards = [this.state.cardDeck.pop(), this.state.cardDeck.pop()];
+    const newCurrCard1 = this.state.cardDeck.pop();
+    const newCurrCard2 = this.state.cardDeck.pop();
+    this.state.player1.currCard = newCurrCard1;
```

Comment:
This way of updating state should always be avoided. Always use this.setState. By right we should treat state as immutable. Otherwise we run into the danger of creating bugs and race conditions in our React app, basically the app might not be able to tell there were state updates and update accordingly.